integration_tests/src/main/python/subquery_test.py

```diff
@@ -53,7 +54,11 @@ def test_scalar_subquery_struct(basic_gen):
         where (select first(ss) from table)['child']['c0'] > ss.child.c0
         ''')
     # struct of array
-    gen = [('ss', StructGen([['arr', ArrayGen(basic_gen)]]))]
+    # Note: The test query accesses the first two elements of the array.  The datagen is set up
+    #       to generate arrays of a minimum of two elements.  Otherwise, the test will fail in ANSI mode.
```

Comment:
There is a loss of test coverage here -- what does Spark do with invalid indices when ANSI is off.  That's what this used to test, and no longer does.  I don't see that covered anywhere else.  IMO this test should remain mostly untouched with the caveat that it explicitly sets ANSI off, and a new test should be added that explicitly sets ANSI on.

Reply:
Ah, I see now that test_scalar_subquery_array_ansi_mode_failures is actually not solely testing for failures but also testing non-ANSI mode (despite what the test name implies). A bit confusing.  So I think we actually are covered, but I still would rather see tests explicitly set what they're assuming -- this is not testing ANSI on for anything but Spark 4, and only because we're getting it by default.  The test shouldl state its assumptions explicitly so we get ANSI/non-ANSI coverage of these cases on all Spark platforms, not just some of them.

Reply:
If we do want to "fix" these tests then we should be explicitly setting the ANSI mode and test both ANSI modes and only set the min length when ANSI is on. 

I suggest that we should just decorate the tests with @disable_ansi_mode in the first pass and then go over all the tests to make sure we handle both ANSI modes

Reply:
These points are taken.  Per offline discussions, the following changes have been incorporated:
1. Not choosing an ANSI mode for `test_scalar_subquery_array` was a mistake.  The test now runs for both.  All previous checks from the non-ANSI test are retained.  In ANSI mode, only valid index values are tested.
2. The separate test `test_scalar_subquery_array_ansi_mode_failures` checks for the exceptions thrown in ANSI mode, on invalid index values.  

I think this flows a little better, too.  Thank you for the suggestion.